core/class.go

```diff
@@ -46,8 +57,270 @@ func (c *Cairo0Class) Version() uint64 {
 	return 0
 }
 
+//nolint:funlen
 func (c *Cairo0Class) Hash() (*felt.Felt, error) {
-	return cairo0ClassHash(c)
+	definition, err := makeDeprecatedVMClass(c)
+	if err != nil {
+		return nil, err
+	}
+
+	var program Program
+	err = json.Unmarshal(definition.Program, &program)
+	if err != nil {
+		return nil, err
+	}
+
+	externalEntryPointElements := make([]*felt.Felt, 0, len(definition.EntryPoints.External)*2)
+	l1HandlerEntryPointElements := make([]*felt.Felt, 0, len(definition.EntryPoints.L1Handler)*2)
+	constructorEntryPointElements := make([]*felt.Felt, 0, len(definition.EntryPoints.Constructor)*2)
+	builtInsHashElements := make([]*felt.Felt, 0, len(program.Builtins))
+	dataHashElements := make([]*felt.Felt, 0, len(program.Data))
+
+	// Use goroutines to parallelize hash computations
+	var wg sync.WaitGroup
+	var externalEntryPointHash, l1HandlerEntryPointHash, constructorEntryPointHash, builtInsHash, dataHash *felt.Felt
+	var hintedClassHash *felt.Felt
+	var hintedClassHashErr error
+
+	wg.Add(6) //nolint:mnd
+
+	go func() {
+		defer wg.Done()
+		for _, ep := range definition.EntryPoints.External {
+			externalEntryPointElements = append(externalEntryPointElements, ep.Selector, ep.Offset)
+		}
+		externalEntryPointHash = crypto.PedersenArray(externalEntryPointElements...)
```

Comment:
can we rewrite these kind of functions into something like:
```
for _, ep := range definition.EntryPoints.External {
   externalEntryPointsDigest.Update(ep.Selector, ep.Offset)
}
```
and later on call `externalEntryPointsDigest.Finish()` at the very bottom of this method?

Pros: we don't allocate additional memory
Cons: if there is a mistake potentially we waste more resources calculating digest